### PR TITLE
fix(ClientRequest): spread object args to support "follow-redirects"

### PR DESCRIPTION
**Pull request by kettanaito**

### CODE REVIEW

File: src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.ts [hunk 1]
```diff
@@ -210,7 +210,7 @@ export function normalizeClientRequestArgs(
   // Handle a given "RequestOptions" object as-is
   // and derive the URL instance from it.
   else if (isObject(args[0])) {
-    options = args[0] as any
+    options = { ... args[0] as any }
```
Comment (kettanaito) [inline src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.ts:213]: The actual fix.

**Comment by kettanaito** [comment]
@mikicho, thanks for fixing this! Can you please give this a quick review to be sure I moved all the changes correctly?

File: src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.test.ts [hunk 2]
```diff
@@ -274,7 +274,7 @@ it('handles [RequestOptions, callback] input', () => {
   expect(url.href).toEqual('https://mswjs.io/resource')
 
   // Request options must be preserved.
-  expect(options).toEqual(initialOptions)
+  expect(options).toMatchObject(initialOptions)
```
Comment (kettanaito) [inline src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.test.ts:277]: With the fix, the actual options object will contain _more_ things, like `_defaultAgent`. I suppose I good thing!

**Comment by mikicho** [comment]
LGTM

**Pull request merged**